src/network.py

```diff
@@ -171,4 +171,75 @@ def forward(self, x):
     # x = self.relu(x) 
     x = self.output(x)
     x = self.sigmoid(x[0])
-    return x 
\ No newline at end of file
+    return x 
+  
+
+#SKIP CONNECTION 
+#Custom Function to write skip connection 
+
+class modifiedSkipConnection(nn.Module):
+    
+    def __init__(self, in_features = 2, out_features = 1, bias=False):
+        super().__init__()
+        self.in_features = in_features
+        self.out_features = out_features
+        #Only 1 weight as a parameter 
+        self.weight = torch.nn.Parameter(torch.Tensor(1,1))
+        if bias:
+            self.bias = torch.nn.Parameter(torch.Tensor(out_features))
+        else:
+            self.register_parameter('bias', None)
+        self.reset_parameters()
+
+    #Initialise the weights  
+    def reset_parameters(self):
+        torch.nn.init.kaiming_uniform_(self.weight, a=math.sqrt(5))  
+        
+    def forward(self, input):
+        x, y = input.shape
+        if y != self.in_features:
+            print(f'Wrong Input Features. Please use tensor with {self.in_features} Input Features')
+            return 0
+        output = torch.Tensor(x,1)
+        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+        for i in range(x):
+          w = torch.nn.sigmoid(self.weight)
```

Comment:
w = torch.nn.sigmoid(self.weight)
this can be defined outside the for loop, since this is constant inside the for loop 

but o.w looks good

Reply:
Done